# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mozglue/tests/TestBaseProfiler.cpp
# Commit: a01db300cf22
# Full Hash: a01db300cf221298a1937d7ba6c82d6466526353
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2020-10-17 09:14:56
# Regressor Bug: 1670954
# File Overlap Count: 5
# Description:
#   Bug 1670954 - profiler_main_thread_id() and profiler_is_main_thread() - r=gregtatum
#   
#   These functions will be useful to get the main thread id, or check if we're in it, in some public code (e.g., markers).
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93735
# ==============================================================================

diff -r c0a3393e4a80 -r a01db300cf22 mozglue/tests/TestBaseProfiler.cpp
--- a/mozglue/tests/TestBaseProfiler.cpp	Fri Oct 16 21:06:13 2020 +0000
+++ b/mozglue/tests/TestBaseProfiler.cpp	Fri Oct 16 22:06:53 2020 +0000
@@ -72,6 +72,31 @@
 
 using namespace mozilla;
 
+void TestProfilerUtilities() {
+  printf("TestProfilerUtilities...\n");
+
+  // We'll assume that this test runs in the main thread (which should be true
+  // when called from the `main` function).
+  const int mainThreadId = mozilla::baseprofiler::profiler_current_thread_id();
+
+  MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_main_thread_id() ==
+                     mainThreadId);
+  MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_is_main_thread());
+
+  std::thread testThread([&]() {
+    const int testThreadId =
+        mozilla::baseprofiler::profiler_current_thread_id();
+    MOZ_RELEASE_ASSERT(testThreadId != mainThreadId);
+
+    MOZ_RELEASE_ASSERT(mozilla::baseprofiler::profiler_main_thread_id() !=
+                       testThreadId);
+    MOZ_RELEASE_ASSERT(!mozilla::baseprofiler::profiler_is_main_thread());
+  });
+  testThread.join();
+
+  printf("TestProfilerUtilities done\n");
+}
+
 void TestPowerOfTwoMask() {
   printf("TestPowerOfTwoMask...\n");
 
@@ -3209,6 +3234,7 @@
 }
 
 void TestProfilerDependencies() {
+  TestProfilerUtilities();
   TestPowerOfTwoMask();
   TestPowerOfTwo();
   TestLEB128();